File: 349-intersection-of-two-arrays/349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> set(nums1.begin(), nums1.end());
        vector<int> res;
        for (int i : nums2)
            if (set.erase(i))
                res.push_back(i);
        return res;
    }
};

File: 349-intersection-of-two-arrays/349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();
        Set<Integer> ans = new HashSet<>();
        
        for(int i : nums1)
            set.add(i);
        
        for(int i: nums2)
            if(set.contains(i))
                ans.add(i);
        
        int[] res = new int[ans.size()];
        
        int i = 0;
        
        for(Integer it: ans)
            res[i++] = it;
        
        return res;
    }
}